keps/sig-auth/4193-tokenrequest-requester-info-enhancements/README.md

```diff
@@ -0,0 +1,596 @@
+
+Enabling the feature gate will cause additional information to be stored/persisted into service account JWTs, as well
+as new audit annotations being recorded in the audit log. This is all purely additive, so no changes to existing
+features, schemas or fields are expected.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes. Future tokens will then not embed this information. Any existing issued tokens **will** still have this
+information embedded however.
+
+If these fields are deemed to be problematic for other systems interpreting these tokens, users will need to re-issue
+these tokens before presenting them elsewhere.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+Future tokens will once again include this information/no adverse effects.
+
+###### Are there any tests for feature enablement/disablement?
+
+Yes (as noted above in the test plan)
+
+### Rollout, Upgrade and Rollback Planning
+
+Rolling this out will be done by enabling the feature flag on all control plane hosts.
+This should not have any issues/affect during upgrades.
+Rollback is done by removing/disabling the feature gate.
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+If somehow the tokens issued during the time where the feature was enabled were to be rejected, these tokens would
+need to be re-requested.
+
+This isn't so much a failure of the rollback, rather it would be a bug in the actual feature (hence feature gating).
+
+During a skew where hosts have different enablement, users would get tokens with varying content if they made multiple
+requests. These tokens will all be usable against all apiservers though, so there won't be issues due to skew.
+
+###### What specific metrics should inform a rollback?
+
+* `authentication_attempts`
+* `authorization_attempts_total`
+* `serviceaccount_valid_tokens_total`
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+<!--
+Describe manual testing that was done and the outcomes.
+Longer term, we may want to require automated upgrade/rollback tests, but we
+are missing a bunch of machinery and tooling and can't do that now.
+-->
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No.
+
+### Monitoring Requirements
+
+N/A
+
+<!--
+This section must be completed when targeting beta to a release.
+
+For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field.
+-->
+
+###### How can an operator determine if the feature is in use by workloads?
+
+For the node info part, using the TokenRequest API and inspecting the contents of the issued JWTs for a token bound to a Pod.
+For JTIs, using the TokenRequest API and then inspecting the contents of the issued JWT for any ServiceAccount token.
```

Comment:
You should also be able to see this just by looking at the metrics of the cluster.

Reply:
Which metrics/what dyou mean here? Inspecting the metrics around successful/failed authentication attempts?